File: examples/csharp/Tutorial34.cs
/* ---------------------------------------------------------------------
 * Tutorial 34
 * 
 * This tutorial shows how to import Excel to DataSet in C#. The data 
 * is imported from the active sheet of the Excel file (the Excel file 
 * generated in Tutorial 09).
 * ------------------------------------------------------------------ */

using System;
using System.IO;
using System.Data;
using EasyXLS;

public class Tutorial34
{

	[STAThread]
	static void Main() 
	{
		Console.WriteLine("Tutorial 34\n-----------\n");

        // Create an instance of the class that imports Excel files
		ExcelDocument workbook = new ExcelDocument();

        // Import Excel file to DataSet
		Console.WriteLine("Reading file C:\\Samples\\Tutorial09.xlsx.\n");
		DataSet ds = workbook.easy_ReadXLSXActiveSheet_AsDataSet("C:\\Samples\\Tutorial09.xlsx");

        // Display imported DataSet values
		DataTable dt = ds.Tables[0];
		for (int row=0; row<dt.Rows.Count; row++)
			for (int column=0; column<dt.Columns.Count; column++)
				Console.WriteLine("At row " + (row + 1) + ", column " + (column + 1) +
					" the value is '" + dt.Rows[row].ItemArray[column] + "'");

		Console.Write("\nPress Enter to Exit...");
		
		// Dispose memory
		workbook.Dispose();

		Console.ReadLine();
	}
}

File: examples/java/Tutorial34.java
//package testexceljava;

import java.io.FileInputStream;
import java.sql.ResultSet;
import EasyXLS.*;

/*------------------------------------------------------------------------
 | Tutorial 34
 | 
 | This tutorial shows how to import Excel to ResultSet in Java. The data 
 | is imported from the active sheet of the Excel file (the Excel file 
 | generated in Tutorial 09).
 -----------------------------------------------------------------------*/

public class Tutorial34 {

  public static void main(String[] args) {
    try {
      System.out.println("Tutorial 34");
      System.out.println("----------");

      // Create an instance of the class that imports Excel files
      ExcelDocument workbook = new ExcelDocument();

      // Import Excel file to ResultSet
      System.out.println("Reading file C:\\Samples\\Tutorial09.xlsx.");
      FileInputStream file = new FileInputStream("C:\\Samples\\Tutorial09.xlsx");
      ResultSet rs = workbook.easy_ReadXLSXActiveSheet_AsResultSet(file);
      
	  // Display imported ResultSet values
      int columnCount = rs.getMetaData().getColumnCount();
      int row = 0;
      while (rs.next()){
        for (int column=1; column<=columnCount; column++)
          System.out.println("At row " + (row + 1) + ", column " + (column) +
                             " the value is '" + rs.getString(column) + "'");
        row++;
      }

      // Dispose memory
      workbook.Dispose();
    }
    catch (Exception ex) {
      ex.printStackTrace();
    }
  }
}
